File: src/main/java/ru/trukhmanov/twochairsbackend/entity/Question.java
package ru.trukhmanov.twochairsbackend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "questions")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name="deck_id", nullable=false)
    private Long deckId;

    @Column(name="option_a", nullable=false)
    private String optionA;

    @Column(name="option_b", nullable=false)
    private String optionB;

    @Column(name="is_active", nullable=false)
    private boolean active;
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/entity/Question.kt
package ru.trukhmanov.twochairsbackend.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "questions")
class Question(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "option_a", nullable = false)
    var optionA: String = "",

    @Column(name = "option_b", nullable = false)
    var optionB: String = "",

    @Column(name = "is_active", nullable = false)
    var active: Boolean = false
)
